src/bigml/api/anomaly_score.clj

```diff
@@ -0,0 +1,136 @@
+;; http://www.apache.org/licenses/LICENSE-2.0
+
+(ns bigml.api.anomaly-score
+  "Functions for scoring with BigML anomaly detectiors.
+     https://bigml.com/api/anomalyscores"
+  (:require (bigml.api [core :as api]
+                       [utils :as utils]))
+  (:refer-clojure :exclude [list]))
+
+(defn create
+  "Creates an anomaly detector from an anomaly detector. A detector
+  can be represented through its id (e.g., `anomaly/123123`), or a map
+  as returned by `get` or `list`.
+
+  The inputs may either be a map (field ids to values), or a sequence
+  of the inputs fields in the order they appeared during training.
+
+  Accepts the optional creation parameters described in the BigML API docs:
+     https://bigml.com/api/anomalyscores#as_anomaly_score_arguments
+
+  HTTP response information is attached as meta data. Exceptions are thrown
+  on failure unless :throw-exceptions is set to false (default is true),
+  in which case the HTTP response details are returned as a map on failure."
+  [anomaly-detector inputs & params]
+  (utils/create :target :anomalyscore
+                :origin [:anomaly anomaly-detector]
+                :params params
+                :inputs inputs))
+
+(defn list
+  "Retrieves a list of anomaly scores. The optional parameters can include
+  pagination and filtering options detailed here:
+     https://bigml.com/api/anomalyscores#as_listing_anomaly_scores
+
+  Pagination details are returned as meta data attached to the list, along
+  with the HTTP response information. Exceptions are thrown on failure unless
+  :throw-exceptions is set to false (default is true), in which case the
+  HTTP response details are returned in a map on failure."
+  [& params]
+  (apply api/list :anomalyscore params))
+
+(defn- predicate-fn [{:keys [field op value] :as pred} id->index]
+  (if (true? pred)
+    (constantly true)
+    (let [index (id->index field)
+          in-set (when (#{"in" "in*"} op) (set value))
+          op-fn (case op
+                  "=" #(and % (= % value))
+                  "=*" #(or (nil? %) (= % value))
+                  "!=" #(not= % value)
+                  ">" #(and % (> % value))
+                  ">*" #(or (nil? %) (> % value))
+                  ">=" #(and % (>= % value))
+                  ">=*" #(or (nil? %) (>= % value))
+                  "<" #(and % (< % value))
+                  "<*" #(or (nil? %) (< % value))
+                  "<=" #(and % (<= % value))
+                  "<=*" #(or (nil? %) (<= % value))
+                  "in" #(and % (boolean (in-set %)))
+                  "in*" #(or (nil? %) (boolean (in-set %))))]
+      (fn [inputs]
+        (op-fn (inputs index))))))
+
+(defn- predicates-fn [predicates id->index]
+  (let [pred-fns (mapv #(predicate-fn % id->index) predicates)
+        pred-count (count predicates)]
+    (fn [inputs]
+      (loop [i 0]
+        (if (and (< i pred-count) ((pred-fns i) inputs))
+          (recur (inc i))
+          i)))))
```

Comment:
I originally did just that. But liked you guessed, it's slower than `loop`. The loops here and in `node-fn` are used so frequently that the time savings were worth it.